node/web/api/booksApiController.js

```diff
@@ -26,5 +25,82 @@ module.exports.factory = function (router, repo, exceptions) {
         });
     });
 
+    router.post('/api/product', function (req, res) {
+      var count = "";
+      var bookTitle = "";
+      repo.get(req.query.product, function (err, book) {
+            if (err) {
+                exceptions.throwException(err);
+                res.status(400).end();
+                return;
+            }
+            var email = "";
+            console.log("Book title: " + book.title);
+            if(!req.cookies.email)
+            {
+
+              email = "Guest";
+            }
+            else {
+              {
+                email = req.cookies.email;
+              }
+            }
+            ordersRepo.updateCart(email,book, function (err, user){
+              if (err){
+                res.status(400).end();
+                return;
+              }
+              });
+              ordersRepo.getCount(email, function (err,doc) {
+                  if (err) {
+                    res.status(400).end();
+                    return;
+                  }
+                  count = doc.total_quantity;
+                  res.send(String(count));
+            });
+        });
+    });
+    router.get('/api/count', function (req, res) {
+
+      var count = "";
+      var bookTitle = "";
+      var email = "";
+              if(!req.cookies.email)
+              {
+                email = "Guest";
+                ordersRepo.remove(email, function (err,doc) {
+                    if (err) {
+                      res.status(400).end();
+                      return;
+                    }
+                    else {
+                      var authCookieExpiryDurationMinutes = 43200, // 30 days
+                          maxAge = authCookieExpiryDurationMinutes * 60 * 1000;
+                      res.cookie('email', email, { maxAge: maxAge, httpOnly: false });
```

Comment:
hi Professor, 
So the logic that we thought was like, if there is no cookies saved (ie either the user clears up the browser cookie or click on logout) then whenever a user directly lands up on the home page, the getCount request should clear up the "Guest" entry from the Orders Table and create a new entry with count = 0 and then treat him as a new Guest user, also therefore we set the cookie to "Guest" so that further calls to the database can use it.